include/s3select_oper.h

```diff
@@ -1417,34 +1432,35 @@ class logical_operand : public base_statement
       throw base_s3select_exception("missing operand for logical ", base_s3select_exception::s3select_exp_en_t::FATAL);
     }
     value a = l->eval();
+    value b = r->eval();
```

Comment:
AND
in case left side is null  , no need to continue(no need to evaluate the right side)

the same for left side is false 

OR
left side is true , result is true (no need to evaluate the right side)

Reply:
I didn't understand the first point. For AND, If left side is null, It will return false if the right side is false. So, we need to evaluate both unless one of them is false.

Reply:
> I didn't understand the first point. For AND, If left side is null, It will return false if the right side is false. So, we need to evaluate both unless one of them is false.

yes, on /AND/ if left=false it return false , no matter what is right side (saving CPU)
on /OR/ if left=true it return true, no matter what is the right side.